### ISSUE TITLE
Private Component Members Are Accessible in Template if Done From Within a FormGroup

### ISSUE THREAD

**Issue by mylifeandcode** [angular/angular#46333]
### Which @angular/* package(s) are the source of the bug?

forms

### Is this a regression?

No

### Description

Component members which are declared private can be accessed from the component's template if done so within a FormGroup.

For example, I have this form which is bound to a FormGroup:

```
<form *ngIf="workoutForm && !loading" [formGroup]="workoutForm" novalidate>
```

Within this form, I am able to access a private method from my component:

```
<wt-exercise-list-mini (exerciseSelected)="addExercise($event)"></wt-exercise-list-mini>
```

I wouldn't expect this to compile, but it does. _However_, if I move that line outside of the form with the FormGroup binding, it results in this error:

```
Property 'addExercise' is private and only accessible within class 'WorkoutEditComponent'.
```

This isn't a new issue and has been like this for at least a few major versions.

### Please provide a link to a minimal reproduction of the bug

https://github.com/mylifeandcode/workouttracker/blob/master/WorkoutTracker.UI/workout-tracker/src/app/workouts/workout-edit/workout-edit.component.html

### Please provide the exception or error you saw

No error if private member is accessed from within a form bound to a FormGroup. If member is accessed from _outside_ of a form bound to a FormGroup, it results in "Property 'addExercise' is private and only accessible within class 'WorkoutEditComponent'."

### Please provide the environment you discovered this bug in (run `ng version`)

Angular CLI: 14.0.1
Node: 16.15.0
Package Manager: npm 8.3.0
OS: win32 x64

Angular: 14.0.1
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1400.1
@angular-devkit/build-angular   14.0.1
@angular-devkit/core            14.0.1
@angular-devkit/schematics      14.0.1
@angular/cdk                    13.3.3
@schematics/angular             14.0.1
rxjs                            6.6.7
typescript                      4.7.3

### Anything else?

Full repo containing example can be found at https://github.com/mylifeandcode/workouttracker .
Angular code is located within workouttracker\WorkoutTracker.UI\workout-tracker

**Comment by JoostK** [comment]
It looks like you're compiling without either `strictTemplates` or `fullTemplateTypeCheck`, as I don't see any `angularCompilerOptions` configuration in any of the `tsconfig.json`/`tsconfig.app.json` files. This would mean that type-checking is fairly loose, in particular it means that anything within a template (also those created using structural directives such as `*ngIf`) will not be checked at all. I suspect that this is the case here.

**Comment by mylifeandcode** [comment]
Thanks @JoostK for the response! I've confirmed that when I remove the *ngIf from that div, I get the error -- I thought the FormGroup binding was the culprit, but was incorrect.

Thanks for the tip about the _strictTemplates_ and _fullTemplateTypeCheck_ settings. This is an old project that I began years with Angular 4. I've been keeping the version of Angular up-to-date but learned only recently that the _tsconfig.json_ settings have remained the same. I made some modifications to them not too long ago to bring it more up-to-date with what a _tsconfig.json_ file from a brand new Angular CLI app looks like, but wasn't aware of the settings you mentioned. Thanks for the information -- I'll be updating that file again in the near future! :)
